alacritty/src/renderer/solidrect.rs

```diff
@@ -0,0 +1,200 @@
+use alacritty_terminal::term::SizeInfo;
+
+use crate::gl;
+use crate::gl::types::*;
+use crate::renderer::rects::RenderRect;
+use crate::renderer::RectShaderProgram;
+
+use std::mem::size_of;
+use std::ptr;
+
+const MAX_U16_INDICES: usize = 65536;
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+struct Rgba {
+    r: u8,
+    g: u8,
+    b: u8,
+    a: u8,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+struct Vertex {
+    // TODO these can certainly be i16
+    x: f32,
+    y: f32,
+    color: Rgba,
```

Comment:
In general, we tend to inline `Rgba` into such things, so just having `r: u8`, `g: u8`, etc, like we have on instance data to pass the same vec4 seems more reasonable, otherwise we should refactor the way we  store such things on `InstanceData` to use new `Rgba` struct, and not just in `RectRenderer`.

Reply:
clippy doesn't like this: `error: 6 bindings with single-character names in scope` :(
```
let x = (rect.x - center_x) / center_x;
let y = -(rect.y - center_y) / center_y;
let width = rect.width / center_x;
let height = rect.height / center_y;
let r = rect.color.r;
let g = rect.color.g;
let b = rect.color.b;
let a = (rect.alpha * 255.) as u8;

// Append rect vertices.
self.vertices.extend_from_slice(&[
    Vertex { x, y: y - height, r, g, b, a },
    Vertex { x: x + width, y, r, g, b, a },
    Vertex { x: x + width, y: y - height, r, g, b, a },
]);
```

Reply:
Locally disabling it here should be an option. I don't see a reason for this lint, but I'd have to check it out myself since I've never seen it before.